src/test/java/SpendingAnalyticsTest.java

```diff
@@ -158,4 +160,90 @@ public void storeWithNullPriceNotIncludedInAnalytics() {
     Assert.assertEquals(1, analytics.getStoreAnalytics().size());
     Assert.assertTrue(analytics.getStoreAnalytics().containsKey("walmart"));
   }
+
+  @Test
+  public void categoryAnalyticsWithUniqueCategoryNames() {
+    // Receipts categories:
+    // candy: $26.12, drink: $26.12, cappuccino: $14.51, food: $14.51
+
+    ImmutableSet<Entity> receipts =
+        new ImmutableSet.Builder<Entity>()
+            .add(TestUtils.createEntity(USER_ID, TIMESTAMP, BLOB_KEY, IMAGE_URL,
+                /* price = */ 26.12, STORE, ImmutableSet.of("candy", "drink"), RAW_TEXT))
+            .add(TestUtils.createEntity(USER_ID, TIMESTAMP, BLOB_KEY, IMAGE_URL,
+                /* price = */ 14.51, STORE, ImmutableSet.of("cappuccino", "food"), RAW_TEXT))
+            .build();
+
+    SpendingAnalytics analytics = new SpendingAnalytics(receipts);
+
+    Assert.assertEquals(26.12, analytics.getCategoryAnalytics().get("candy"), ERROR_THRESHOLD);
+    Assert.assertEquals(26.12, analytics.getCategoryAnalytics().get("drink"), ERROR_THRESHOLD);
+    Assert.assertEquals(14.51, analytics.getCategoryAnalytics().get("cappuccino"), ERROR_THRESHOLD);
+    Assert.assertEquals(14.51, analytics.getCategoryAnalytics().get("food"), ERROR_THRESHOLD);
```

Comment:
Based on [tott/538](https://drive.google.com/file/d/1uzaIjN8oLqdVL6Sp0VE7ZnBoehEfg_rz/view), it might be useful to save the price values in local variables.